sql/server.go

```diff
@@ -214,11 +241,16 @@ func (s *Server) exec(req driver.Request) (driver.Response, error) {
 		if err := plan.Err(); err != nil {
 			return resp, err
 		}
-
 		resp.Results = append(resp.Results, result)
 	}
 
 	// Update session state.
+	if planner.txn != nil {
+		planner.session.Txn = client.GetProtoFromTxn(planner.txn)
+		log.Infof("Send txn back to client %v", planner.session.Txn)
+	} else if planner.session.Txn != nil {
+		panic("session contains a transaction for an inactive transaction.")
```

Comment:
This panic message is confusing. Perhaps `session contains a transaction proto but there is no active transaction`.